### ISSUE TITLE
PATH warning and `gem install` not installing to $GEM_HOME

### ISSUE THREAD

**Issue by dpiddy** [rubygems/rubygems#1310]
When installing a gem, I'm getting this warning:

```
% gem install bundler
WARNING:  You don't have /Users/dan.peterson/.gem/ruby/2.2.0/bin in your PATH,
	  gem executables will not run.
Successfully installed bundler-1.10.5
1 gem installed
```

`/Users/dan.peterson/.gem/ruby/2.2.0/bin` is not mentioned in my `$GEM_HOME` or `$GEM_PATH` but the gem is in fact being installed to `/Users/dan.peterson/.gem/ruby/2.2.0/gems/bundler-1.10.5`.

Even if I do prepend my `$PATH` with that directory I don't get the new version:

```
% which bundle  
/Users/dan.peterson/.gem/ruby/2.2.0/bin/bundle
% bundle version
Bundler version 1.9.6
```

I have to also change my `$GEM_PATH` to get it working:

```
% export GEM_PATH=/Users/dan.peterson/.gem/ruby/2.2.0
% bundle version                                     
Bundler version 1.10.5
```

I am using chruby but given my `$PATH`, `$GEM_HOME`, and `$GEM_PATH`, I'm not sure if it's causing the trouble. Here is my `gem env` output:

```
RubyGems Environment:
  - RUBYGEMS VERSION: 2.4.8
  - RUBY VERSION: 2.2.2 (2015-04-13 patchlevel 95) [x86_64-darwin14]
  - INSTALLATION DIRECTORY: /Users/dan.peterson/.gem/ruby/2.2.2
  - RUBY EXECUTABLE: /Users/dan.peterson/.rubies/ruby-2.2.2/bin/ruby
  - EXECUTABLE DIRECTORY: /Users/dan.peterson/.gem/ruby/2.2.2/bin
  - SPEC CACHE DIRECTORY: /Users/dan.peterson/.gem/specs
  - SYSTEM CONFIGURATION DIRECTORY: /Users/dan.peterson/.rubies/ruby-2.2.2/etc
  - RUBYGEMS PLATFORMS:
    - ruby
    - x86_64-darwin-14
  - GEM PATHS:
     - /Users/dan.peterson/.gem/ruby/2.2.2
     - /Users/dan.peterson/.rubies/ruby-2.2.2/lib/ruby/gems/2.2.0
  - GEM CONFIGURATION:
     - :update_sources => true
     - :verbose => true
     - :backtrace => false
     - :bulk_threshold => 1000
     - :sources => ["https://rubygems.org/"]
     - "install" => "--no-document --user-install"
  - REMOTE SOURCES:
     - https://rubygems.org/
  - SHELL PATH:
     - /Users/dan.peterson/.gem/ruby/2.2.2/bin
     - /Users/dan.peterson/.rubies/ruby-2.2.2/lib/ruby/gems/2.2.0/bin
     - /Users/dan.peterson/.rubies/ruby-2.2.2/bin
     - /usr/local/heroku/bin
     - /Users/dan.peterson/bin
     - /usr/local/go/bin
     - /usr/local/bin
     - /usr/local/bin
     - /usr/bin
     - /bin
     - /usr/sbin
     - /sbin
     - /usr/local/go/bin
     - /usr/local/MacGPG2/bin
     - /Users/dan.peterson/Projects/go/bin
```

**Comment by dpiddy** [comment]
It's because I have `--user-install` in my `install` options which uses [`Gem.user_dir`](https://github.com/rubygems/rubygems/blob/v2.4.8/lib/rubygems/installer.rb#L137) which uses [`ruby_version`](https://github.com/rubygems/rubygems/blob/v2.4.8/lib/rubygems/defaults.rb#L77) instead of, say, `RUBY_PROGRAM_VERSION` like I was expecting.

I'm not sure if this is a bug or not. It would at least be nice if `gem env` showed me what the `--user-install` destination would be.

**Comment by tenderlove** [comment]
@dpiddy I think `INSTALLATION DIRECTORY` in your `gem env` output is showing that it will install to your user directory.  Was there something else you had in mind?

**Comment by dpiddy** [comment]
@tenderlove I'm guessing `INSTALLATION DIRECTORY` comes from `$GEM_HOME` which I'd expect (note it uses 2.2.2 in there). What I think is missing from `gem env` is where things will be installed when using `--user-install`, which apparently is not `$GEM_HOME` since for me it was a directory with 2.2.0 in it.

**Comment by tenderlove** [comment]
Ahhh, I see. So you have `$GEM_HOME` set, but that's different than `--user-install`.  I understand now.  You want to see the location for `--user-install` were you to have provided it.

**Comment by dpiddy** [comment]
Yeah. I started debugging this by doing something like `gem env | grep 2.2.0` to see where that path was coming from. If I saw it listed as `USER INSTALLATION DIRECTORY` it probably would have helped me figure out what was going on sooner.

**Comment by djberg96** [comment]
Merged PR #1343. Going to close this one.
